### DISCUSSION TITLE
Reading(printing) an analogvalueobject

### DISCUSSION THREAD

**Discussion by vukmirovic98** [JoelBender/bacpypes#400] [General]
I am an absolute beginner in BACnet and bacpypes and I absolutely don't have an idea what I am doing (this is an assignment for me to try and figure out).
So I have the following code: https://pastebin.com/VMR0Cpep
What I am trying to do for now is just printing the analog 'presentValue'  and that's it.

In the future, I would probably want to write some analog values to but will create a issue if I need help with it.

**Comment by JoelBender** [comment]
(Note that I've moved this over to a discussion rather than an issue.)

1) the expression `_debug == 1 or 2` will always be true!  If `_debug == 1` then the expression is `True`, but if it is some other value then the expression is `2` which is non-zero and therefore `True`.  You can change this to `_debug == 1 or _debug == 2` or `_debug in (1, 2)` or `_debug >= 1`, etc.


2) I do not have a **misty** environment so I switched the application back to a **BIPSimpleApplication** and commented out the `mstp_args` pieces.
3) Changing `_debug = 2` enables the property monitor code.
4) I added the print functions to display the present value and change it and you can see that your `something_changed()` function is called.

See the attached file (as text).
[VMR0Cpep.py.txt](https://github.com/JoelBender/bacpypes/files/6362218/VMR0Cpep.py.txt)
